package/kernel-osmc/initramfs-src/init

```diff
@@ -83,8 +83,37 @@ for option in $(/bin/busybox cat /proc/cmdline); do
 	    ;;
 	  root=*)
 		OPTION_ROOT=$(echo "${option#*=}" | /bin/busybox sed 's/"//g')
+		ROOT_STR=$OPTION_ROOT
 		PARTUUID=$(echo "$OPTION_ROOT" | /bin/busybox grep "^PARTUUID=" | /bin/busybox sed 's/^PARTUUID=//g')
-		if [ -n "$PARTUUID" ]; then OPTION_ROOT=$(/bin/busybox readlink -f /dev/disk/by-partuuid/${PARTUUID}); fi
+		UUID=$(echo "$OPTION_ROOT" | /bin/busybox grep "^UUID=" | /bin/busybox sed 's/^UUID=//g')
+		if ! [ -z "$UUID$PARTUUID" ]; then
+			# it takes a while for USB partitions to show up
+			count=12
+			OPTION_ROOT=
+			while [ $count -ge 0 -a -z "$OPTION_ROOT" ]; do
+				echo $count seconds
+				echo $(/bin/busybox blkid)
+				if [ -n "$PARTUUID" ]; then OPTION_ROOT=$(/bin/busybox findfs "PARTUUID=${PARTUUID}"); fi
+				if [ -n "$UUID" ]; then OPTION_ROOT=$(/bin/busybox findfs "UUID=${UUID}"); fi
+				if ! [ -z "$OPTION_ROOT" ]; then break; fi
```

Comment:
Isn't there *always* an option root as defined in init.d/...?
Do you mean to check for ROOT_STR being set instead? That will still also always have a value.

Reply:
Default OPTION_ROOT is defined in init and in init.d.  Not sure what takes precedence but the one in cmdline.txt should always override, but only if it exists, which is what I'm trying to establish.

I put ROOT_STR there only so I can print it if it can't be found.